plugins/modules/foreman_hostgroup.py

```diff
@@ -118,31 +211,99 @@ def main():
+        if 'compute_resource' in entity_dict:
+            entity_dict['compute_resource'] = module.find_resource_by_name('compute_resources', name=entity_dict['compute_resource'], failsafe=False, thin=True)
+
+        if 'compute_profile' in entity_dict:
+            entity_dict['compute_profile'] = module.find_resource_by_name('compute_profiles', name=entity_dict['compute_profile'], failsafe=False, thin=True)
+
+        if 'domain' in entity_dict:
+            entity_dict['domain'] = module.find_resource_by_name('domains', name=entity_dict['domain'], failsafe=False, thin=True)
+
+        if 'subnet' in entity_dict:
+            entity_dict['subnet'] = module.find_resource_by_name('subnets', name=entity_dict['subnet'], failsafe=False, thin=True)
+
+        if 'subnet6' in entity_dict:
+            entity_dict['subnet6'] = module.find_resource_by_name('subnets', name=entity_dict['subnet6'], failsafe=False, thin=True)
+
+        if 'realm' in entity_dict:
+            entity_dict['realm'] = module.find_resource_by_name('realms', name=entity_dict['realm'], failsafe=False, thin=True)
+
         if 'architecture' in entity_dict:
             entity_dict['architecture'] = module.find_resource_by_name('architectures', name=entity_dict['architecture'], failsafe=False, thin=True)
 
         if 'operatingsystem' in entity_dict:
-            entity_dict['operatingsystem'] = module.find_resource_by_name('operatingsystems', name=entity_dict['operatingsystem'], failsafe=False, thin=True)
+            entity_dict['operatingsystem'] = module.find_resource_by_title('operatingsystems', title=entity_dict['operatingsystem'], failsafe=False, thin=True)
 
         if 'media' in entity_dict:
             entity_dict['media'] = module.find_resource_by_name('media', name=entity_dict['media'], failsafe=False, thin=True)
 
         if 'ptable' in entity_dict:
             entity_dict['ptable'] = module.find_resource_by_name('ptables', name=entity_dict['ptable'], failsafe=False, thin=True)
 
-    entity = module.find_resource_by_name('hostgroups', name=entity_dict['name'], failsafe=True)
+        if 'environment' in entity_dict:
+            entity_dict['environment'] = module.find_resource_by_name('environments', name=entity_dict['environment'], failsafe=False, thin=True)
+
+        if 'config_groups' in entity_dict:
+            entity_dict['config_groups'] = module.find_resources_by_name('config_groups', entity_dict['config_groups'], failsafe=False, thin=True)
+
+        if 'puppet_proxy' in entity_dict:
+            entity_dict['puppet_proxy'] = module.find_resource_by_name('smart_proxies', name=entity_dict['puppet_proxy'], failsafe=False, thin=True)
+
+        if 'puppet_ca_proxy' in entity_dict:
+            entity_dict['puppet_ca_proxy'] = module.find_resource_by_name('smart_proxies', name=entity_dict['puppet_ca_proxy'], failsafe=False, thin=True)
+
+    entity = module.find_resource('hostgroups', search='title="{}"'.format(build_fqn(name, parent)), failsafe=True)
```

Comment:
Any reason not to use `find_resource_by_title` here?

Reply:
No reason. Changed.